rpcs3/Emu/Cell/SPURecompiler.cpp

```diff
@@ -7749,9 +7749,9 @@ class spu_llvm_recompiler : public spu_recompiler_base, public cpu_translator
 
 			const auto ma = eval(sext<s32[4]>(fcmp_uno(a != fsplat<f32[4]>(0.))));
 			const auto mb = eval(sext<s32[4]>(fcmp_uno(b != fsplat<f32[4]>(0.))));
-			const auto ca = eval(bitcast<f32[4]>(bitcast<s32[4]>(a) & mb));
-			const auto cb = eval(bitcast<f32[4]>(bitcast<s32[4]>(b) & ma));
-			set_vr(op.rt, fm(ca, cb));
+			const auto cx = eval(ma & mb);
+			const auto x = fm(a, b);
```

Comment:
Shouldn't you modify fm function as well? llvm expressions detection relies on this.

Reply:
fm is currently just an unnecessary alias for multiplication operator

Reply:
> Shouldn't you modify fm function as well? llvm expressions detection relies on this.

The only time we look for the fm pattern is in is_input_positive, which should still be working since it only looks for the case when a = b